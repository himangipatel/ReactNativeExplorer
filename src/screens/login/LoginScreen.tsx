import {View, Text, Animated} from 'react-native';
import React, {useEffect, useRef} from 'react';
import {StackNavigationProp} from '@react-navigation/stack';
import {RootStackParamList} from '../../navigation/Navigation';
import {styles} from './style';
import AppTextInput from '../../component/AppTextInput';
import AppButton from '../../component/AppButton';
import useValidateEmailPass from '../../hooks/useValidateIDPass';
import {colors} from '../../utils/colors';
import useLocalize from '../../hooks/useLocalize';
import {screens} from '../../utils/screens';
import {Language, setLanguage} from '../../redux/slices/languageSlice';
import ArrowRight from '../../assets/arrow.svg';
import {TRANSLATION_KEYS} from '../../utils/translations';
import responsive from '../../utils/responsive';
import {useAppDispatch} from '../../redux/store';
import {AppStatusBar} from '../../component/StatusBar';
import useLogin from '../../hooks/useLogin';

interface LoginProps {
  navigation: StackNavigationProp<RootStackParamList, 'Login'>;
}

const LoginScreen = ({navigation}: LoginProps) => {
  const dispatch = useAppDispatch();
  const {translate, local} = useLocalize();
  const translateY = useRef(new Animated.Value(300)).current;
  const {adduser} = useLogin();

  const {
    isValidEmail,
    email,
    validateEmail,
    validatePassword,
    isValidPassword,
    password,
  } = useValidateEmailPass();

  const handleLanguageChange = (language: string) => {
    dispatch(setLanguage(language));
    navigation.replace(screens.splash);
  };

  const isValidData = isValidEmail && isValidPassword;

  const handleLogin = () => {
    adduser(email, password);
    navigation.replace(screens.popularMovies);
  };

  useEffect(() => {
    Animated.timing(translateY, {
      toValue: 0,
      duration: 500,
      useNativeDriver: true,
    }).start();
  }, []);

  const rightButton = () => (
    <ArrowRight
      width={24}
      height={24}
      fill={'red'}
      style={{paddingHorizontal: 20}}
    />
  );

  return (
    <View style={styles.container}>
      <AppStatusBar />
      <View style={styles.topContainer}>
        <View style={styles.languageContainer}>
          <Text
            onPress={() => handleLanguageChange(Language.ENGLISH)}
            style={[
              styles.smallText,
              local == Language.ARABIC ? styles.inActive : styles.active,
            ]}>
            {translate(TRANSLATION_KEYS.ENGLISH)}
          </Text>
          <View style={{width: responsive.pWidth(10)}} />
          <Text
            onPress={() => handleLanguageChange(Language.ARABIC)}
            style={[
              styles.smallText,
              local == Language.ARABIC ? styles.active : styles.inActive,
            ]}>
            {translate(TRANSLATION_KEYS.ARABIC)}
          </Text>
        </View>
        <Animated.View
          style={[styles.topAnimatedView, {transform: [{translateY}]}]}>
          <Text style={styles.smallText}>
            {translate(TRANSLATION_KEYS.APP_TITLE)}
          </Text>
          <Text style={[styles.smallText, styles.loginText]}>
            {translate(TRANSLATION_KEYS.LOGIN_TITLE)}
          </Text>
        </Animated.View>
      </View>

      <View style={styles.bottomContainer}>
        <Animated.View style={{transform: [{translateY}]}}>
          <AppTextInput
            placeholder={translate(TRANSLATION_KEYS.ENTER_EMAIL)}
            keyboardType="email-address"
            value={email}
            onChangeText={text => validateEmail(text)}
            showError={!isValidEmail}
            errorMessage={translate(TRANSLATION_KEYS.ENTER_VALID_EMAIL)}
          />

          <AppTextInput
            secureTextEntry
            placeholder={translate(TRANSLATION_KEYS.ENTER_PASSWORD)}
            value={password}
            onChangeText={text => validatePassword(text)}
            showError={!isValidPassword}
            errorMessage={translate(TRANSLATION_KEYS.ENTER_VALID_PASSWORD)}
          />
        </Animated.View>

        <AppButton
          label={translate(TRANSLATION_KEYS.SIGN_IN)}
          rightIcon={rightButton()}
          disable={!isValidData}
          style={{
            backgroundColor: isValidData ? colors.primary : colors.buttonBg,
          }}
          onPress={handleLogin}
        />
      </View>
    </View>
  );
};

export default LoginScreen;
