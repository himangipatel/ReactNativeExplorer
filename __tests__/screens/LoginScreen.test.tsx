import React from 'react';
import LoginScreen from '../../src/screens/login/LoginScreen';
import {fireEvent, render} from '@testing-library/react-native';
import {renderWithProviders} from '../test_utils';

const mockNavigation: any = {
  replace: jest.fn(),
};
// const mockRoute: any = jest.fn();
// const mockDispatch = jest.fn();

// jest.mock('react-redux', () => ({
// useSelector: jest.fn(),
// useDispatch: () => mockDispatch,
// }));

describe('Login', () => {
  it('renders correctly', () => {
    const loginUI = renderWithProviders(
      <LoginScreen navigation={mockNavigation} />,
    );

    expect(loginUI.toJSON()).toMatchSnapshot();
  });
});
