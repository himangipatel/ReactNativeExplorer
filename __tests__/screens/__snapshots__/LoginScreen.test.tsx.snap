// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders Login correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "black",
        "flex": 1,
        "padding": 40,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingVertical": 100,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          [
            {
              "color": "white",
              "fontSize": 12,
              "fontWeight": "bold",
              "opacity": 0.7,
            },
            {
              "fontWeight": "bold",
              "opacity": 1,
            },
          ]
        }
      >
        ENGLISH
      </Text>
      <View
        style={
          {
            "width": 75,
          }
        }
      />
      <Text
        onPress={[Function]}
        style={
          [
            {
              "color": "white",
              "fontSize": 12,
              "fontWeight": "bold",
              "opacity": 0.7,
            },
            {
              "fontWeight": "bold",
              "opacity": 0.7,
            },
          ]
        }
      >
        ARABIC
      </Text>
    </View>
    <View
      collapsable={false}
      style={
        {
          "flex": 1,
          "justifyContent": "flex-end",
          "transform": [
            {
              "translateY": 300,
            },
          ],
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 12,
            "fontWeight": "bold",
            "opacity": 0.7,
          }
        }
      >
        MOST POPULAR MOVIES OF ALL TIME
      </Text>
      <Text
        style={
          [
            {
              "color": "white",
              "fontSize": 12,
              "fontWeight": "bold",
              "opacity": 0.7,
            },
            {
              "fontSize": 24,
              "opacity": 1,
              "paddingVertical": 30,
            },
          ]
        }
      >
        tell us about your email and password
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "white",
        "flex": 2,
        "justifyContent": "space-between",
        "padding": 10,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "translateY": 300,
            },
          ],
        }
      }
    >
      <View>
        <TextInput
          autoCapitalize="none"
          keyboardType="email-address"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Enter Email ID"
          placeholderTextColor="#ccc"
          style={
            [
              undefined,
              {
                "borderColor": "#ccc",
                "borderRadius": 5,
                "borderWidth": 1,
                "color": "black",
                "margin": 10,
                "padding": 15,
              },
              {
                "textAlign": "left",
              },
            ]
          }
          value=""
        />
      </View>
      <View>
        <TextInput
          autoCapitalize="none"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Enter Password"
          placeholderTextColor="#ccc"
          secureTextEntry={true}
          style={
            [
              undefined,
              {
                "borderColor": "#ccc",
                "borderRadius": 5,
                "borderWidth": 1,
                "color": "black",
                "margin": 10,
                "padding": 15,
              },
              {
                "textAlign": "left",
              },
            ]
          }
          value=""
        />
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignSelf": "flex-start",
          "backgroundColor": "gray",
          "margin": 30,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignContent": "center",
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 20,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 22,
              "padding": 30,
            }
          }
        >
          Login and Continue
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Login renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "black",
        "flex": 1,
        "padding": 40,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingVertical": 100,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          [
            {
              "color": "white",
              "fontSize": 12,
              "fontWeight": "bold",
              "opacity": 0.7,
            },
            {
              "fontWeight": "bold",
              "opacity": 1,
            },
          ]
        }
      >
        ENGLISH
      </Text>
      <View
        style={
          {
            "width": 75,
          }
        }
      />
      <Text
        onPress={[Function]}
        style={
          [
            {
              "color": "white",
              "fontSize": 12,
              "fontWeight": "bold",
              "opacity": 0.7,
            },
            {
              "fontWeight": "bold",
              "opacity": 0.7,
            },
          ]
        }
      >
        ARABIC
      </Text>
    </View>
    <View
      collapsable={false}
      style={
        {
          "flex": 1,
          "justifyContent": "flex-end",
          "transform": [
            {
              "translateY": 300,
            },
          ],
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 12,
            "fontWeight": "bold",
            "opacity": 0.7,
          }
        }
      >
        MOST POPULAR MOVIES OF ALL TIME
      </Text>
      <Text
        style={
          [
            {
              "color": "white",
              "fontSize": 12,
              "fontWeight": "bold",
              "opacity": 0.7,
            },
            {
              "fontSize": 24,
              "opacity": 1,
              "paddingVertical": 30,
            },
          ]
        }
      >
        tell us about your email and password
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "white",
        "flex": 2,
        "justifyContent": "space-between",
        "padding": 10,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "translateY": 300,
            },
          ],
        }
      }
    >
      <View>
        <TextInput
          autoCapitalize="none"
          keyboardType="email-address"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Enter Email ID"
          placeholderTextColor="#ccc"
          style={
            [
              undefined,
              {
                "borderColor": "#ccc",
                "borderRadius": 5,
                "borderWidth": 1,
                "color": "black",
                "margin": 10,
                "padding": 15,
              },
              {
                "textAlign": "left",
              },
            ]
          }
          testID="email-input"
          value=""
        />
      </View>
      <View>
        <TextInput
          autoCapitalize="none"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Enter Password"
          placeholderTextColor="#ccc"
          secureTextEntry={true}
          style={
            [
              undefined,
              {
                "borderColor": "#ccc",
                "borderRadius": 5,
                "borderWidth": 1,
                "color": "black",
                "margin": 10,
                "padding": 15,
              },
              {
                "textAlign": "left",
              },
            ]
          }
          testID="password-input"
          value=""
        />
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignSelf": "flex-start",
          "backgroundColor": "gray",
          "margin": 30,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignContent": "center",
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 20,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 22,
              "padding": 30,
            }
          }
        >
          Login and Continue
        </Text>
      </View>
    </View>
  </View>
</View>
`;
